From c1923135095cf8b8640cc95052e1778b551a475b Mon Sep 17 00:00:00 2001
From: Georges Basile Stavracas Neto <georges.stavracas@gmail.com>
Date: Tue, 22 Aug 2023 18:20:36 -0300
Subject: [PATCH] data: Add gnome-portals.conf

Now that xdg-desktop-portal handles portal implementations using
a well-defined config file, let's ship the GNOME blend of portals
in gnome-portals.conf. Install it under datadir/xdg-desktop-portal,
which usually would resolve to /usr/share/xdg-desktop-portal.

Closes: https://gitlab.gnome.org/GNOME/gnome-session/-/issues/114
---
 data/gnome-portals.conf | 4 ++++
 data/meson.build        | 5 +++++
 2 files changed, 9 insertions(+)
 create mode 100644 data/gnome-portals.conf

diff --git a/data/gnome-portals.conf b/data/gnome-portals.conf
new file mode 100644
index 00000000..198e6445
--- /dev/null
+++ b/data/gnome-portals.conf
@@ -0,0 +1,4 @@
+[preferred]
+default=gnome;gtk;
+org.freedesktop.impl.portal.Access=gnome-shell;gtk;
+org.freedesktop.impl.portal.Secret=gnome-keyring;
diff --git a/data/meson.build b/data/meson.build
index f6d264f8..f37924b8 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -195,3 +195,8 @@ install_data(
   data,
   install_dir: session_pkgdatadir
 )
+
+install_data(
+  'gnome-portals.conf',
+  install_dir: session_datadir / 'xdg-desktop-portal',
+)
-- 
2.42.0

